spec/tasks/themes_spec.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+RSpec.describe "tasks/themes" do
+  before do
+    Rake::Task.clear
+    Discourse::Application.load_tasks
+  end
+
+  describe "themes:update" do
+    let(:initial_repo) do
+      about_json = <<~JSON
+      {
+        "name": "awesome theme",
+        "about_url": "https://www.site.com/about",
+        "license_url": "https://www.site.com/license",
+        "theme_version": "1.0",
+        "minimum_discourse_version": "1.0.0"
+      }
+      JSON
+
+      setup_git_repo("about.json" => about_json)
+    end
+
+    let(:initial_repo_url) do
+      MockGitImporter.register("https://example.com/initial_repo.git", initial_repo)
+    end
+
+    let!(:theme) { RemoteTheme.import_theme(initial_repo_url) }
+
+    around(:each) { |group| MockGitImporter.with_mock { group.run } }
+
+    after { `rm -fr #{initial_repo}` }
+
+    it "should not update the theme if theme settings migration fails" do
+      add_to_git_repo(initial_repo, "migrations/settings/0001-a-migration.js" => <<~JS)
```

Comment:
Can we accompany this migration file with a change to an existing file (e.g. an hbs template or a js file) and then, after the migration fails, assert that the existing file is not changed and the migration file doesn't exist?

Reply:
Good point, let me do that too 👍